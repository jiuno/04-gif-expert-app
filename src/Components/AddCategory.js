import React, { useState } from 'react';
import PropTypes from 'prop-types';


export  const AddCategory = ({SetCategories}) => {
    
        const  [InputValue, setInputValue] = useState('');

        const handleInputChange = (e) => {
            
            setInputValue(e.target.value)
        }
        
        const handleSubmit = (e) => {
            e.preventDefault();
            if (InputValue.trim().length >= 2) {
                SetCategories((cats) => [InputValue, ...cats]); //Esto es categories ahora
                setInputValue(''); 
                
            }


            // console.log('Submit hecho');

        }
        return (
            <form onSubmit= {handleSubmit}>
                <input
                type="text" 
                value={InputValue}
                onChange={handleInputChange}
                
                />
            </form>
        )
    }

AddCategory.propTypes = {
    SetCategories: PropTypes.func.isRequired
}

