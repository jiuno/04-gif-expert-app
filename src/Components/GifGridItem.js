import React from 'react'

export const GifGridItem = ({id, title, url}) => {
    
    // console.log(url);
    
    return (
        <div className="card animate__animated animate__bounceIn ">
            <img src={url} alt="GIF" ></img>
            <hr></hr>
            <h3> {title} </h3>
        </div>
    )
}
