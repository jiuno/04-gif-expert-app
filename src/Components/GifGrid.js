import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    const {data:images, loading} = useFetchGifs( category );

    return (
        <>
        <h3> {category} </h3>

        <div className="Card-Grid">
           
                {loading && <p className="animate__animated animate__heartBeat">Loading...</p> }
                { images.map( (img) =>  (
                <GifGridItem 
                { ...img} //Manda props.img con todas sus propiedades, no solo id, title y url.
                key={img.id}
                />))}

        </div>
        </>
    )
}
