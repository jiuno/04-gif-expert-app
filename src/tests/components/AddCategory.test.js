import { shallow } from "enzyme";
import {AddCategory} from '../../Components/AddCategory'
import React from 'react';



describe('Pruebas para <AddCategory />', () => {
    
    const SetCategories = () => {} //Solo es para poder enviar un objeto que es una funcion y cumplir el Proptype.

    const wrapper = shallow(<AddCategory SetCategories={SetCategories} />)

    test('Debe renderizar <AddCategory', () => {
        
        expect(wrapper).toMatchSnapshot();

    });

    test('debe de cambiar la caja de texto', () => {
        
        const input = wrapper.find('input');
        const value = 'Hola Mundo';
        
        input.simulate('change', {target : {value}});

        const testh3 = wrapper.find('h3');

        expect( testh3.text().trim()).toBe(`Vas a enviar ${value}`);
    })
    
    
})
