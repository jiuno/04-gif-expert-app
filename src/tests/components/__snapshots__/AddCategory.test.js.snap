// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para <AddCategory /> Debe renderizar <AddCategory 1`] = `
<form
  onSubmit={[Function]}
>
  <h3>
    Vas a enviar 
  </h3>
   
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
